tests/Query/TermsTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Solr Client Symfony package.
+ *
+ * (c) ingatlan.com Zrt. <fejlesztes@ingatlan.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace iCom\SolrClient\Tests\Query;
+
+use iCom\SolrClient\Query\Terms;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \iCom\SolrClient\Query\Terms
+ */
+final class TermsTest extends TestCase
+{
+    /** @test */
+    public function it_throws_exception_on_invalid_method(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        Terms::create('id', [])->method('method');
+    }
+
+    /**
+     * @test
+     * @dataProvider provider
+     */
+    public function it_creates_terms_query_string(\Closure $terms, string $expectedTerms): void
+    {
+        $this->assertSame($expectedTerms, (string) $terms());
+    }
+
+    public function provider(): iterable
+    {
+        yield 'field' => [
+            static function (): Terms { return Terms::create('id', []); },
+            '{!terms f=id}',
+        ];
+
+        yield 'method' => [
+            static function (): Terms { return Terms::create('id', [])->method('termsFilter'); },
+            '{!terms f=id method=termsFilter}',
+        ];
+
+        yield 'separator-comma' => [
+            static function (): Terms { return Terms::create('id', [1, 2, 3])->separator(','); },
```

Comment:
Ilyenkor a function és a `()` közé sztm. nem kell space.

Reply:
De amúgy mért closure? :)

Reply:
A szóköz szükséges a php_cs_fixer elvárja, és maradtam a closure-öknél, mikor átírtam, hogy csak egy Term objektumot adjon vissza, akkor a tesztek lefutottak, viszont az infection eltört. Azt írta, hogy ha kiveszi az elfogadott metódusok közül a 'termsFilter'-t, akkor is le tud futni, viszont erre van külön teszt, és el kellett volna emaitt törnie.